From 31faa590984b0bae5e5c9d718e1e12fa66a63cfd Mon Sep 17 00:00:00 2001
From: Cole Robinson <crobinso@redhat.com>
Date: Wed, 29 Jan 2014 17:15:16 +0100
Subject: [PATCH 08/10] virt-install: Support --network source, source_mode,
 target

The latter two bits are needed for macvtap configuration.

Notes:
new tests weren't backported.

(cherry picked from commit 95170e8892ecac6a8762990d4cbef9bac14095b2)
Signed-off-by: Giuseppe Scrivano <gscrivan@redhat.com>
---
 man/en/virt-install.pod    | 4 ++++
 man/en/virt-install.pod.in | 4 ++++
 virtinst/cli.py            | 3 +++
 3 files changed, 11 insertions(+)

diff --git a/man/en/virt-install.pod b/man/en/virt-install.pod
index a0b2efb..39ebc6a 100644
--- a/man/en/virt-install.pod
+++ b/man/en/virt-install.pod
@@ -640,6 +640,10 @@ with a name of C<default>. Use a virtual network if the host has dynamic
 networking (eg NetworkManager), or using wireless. The guest will be
 NATed to the LAN by whichever connection is active.
 
+=item type=direct,source=IFACE[,source_mode=MODE]
+
+Direct connect to host interface IFACE using macvtap.
+
 =item user
 
 Connect to the LAN using SLIRP. Only use this if running a QEMU guest as
diff --git a/man/en/virt-install.pod.in b/man/en/virt-install.pod.in
index 69a6bd3..3f6d23e 100644
--- a/man/en/virt-install.pod.in
+++ b/man/en/virt-install.pod.in
@@ -576,6 +576,10 @@ with a name of C<default>. Use a virtual network if the host has dynamic
 networking (eg NetworkManager), or using wireless. The guest will be
 NATed to the LAN by whichever connection is active.
 
+=item type=direct,source=IFACE[,source_mode=MODE]
+
+Direct connect to host interface IFACE using macvtap.
+
 =item user
 
 Connect to the LAN using SLIRP. Only use this if running a QEMU guest as
diff --git a/virtinst/cli.py b/virtinst/cli.py
index 27a512e..51e5486 100644
--- a/virtinst/cli.py
+++ b/virtinst/cli.py
@@ -1629,6 +1629,9 @@ def parse_network(guest, optstring, dev=None, mac=None):
     set_param("type", "type", net_type)
     set_param("network", "network")
     set_param("bridge", "bridge")
+    set_param("source", "source")
+    set_param("source_mode", "source_mode")
+    set_param("target_dev", "target")
     set_param("model", "model")
     set_param("macaddr", "mac")
 
-- 
1.8.5.3

